Notebook.ipynb

```diff
@@ -136,47 +139,66 @@
+    "As discussed during the oral presentation, and as described in the previous paragraph, the knowledge base is where both the information is stored and the decision making process for the agent takes place. This entity itself is divided in two parts, which can be distinguished by the two different languages in which they're written: Prolog and Python. In fact, Python is used as a wrapper interface to control the Prolog part, thanks to the open source library pyswip.\n",
     "The subtasks in the knowledge base are defined in order of relevance. We have:\n",
     "- getCarrot\n",
     "- feedSteed\n",
     "- getSaddle\n",
     "- applySaddle\n",
-    "- explore (this subtask is chosen when no premise is satisfied for another subtask to start)\n",
+    "- explore (chosen when no premise is satisfied for another subtask to start)\n",
     "- eat\n",
     "- attackEnemy\n",
     "\n",
-    "After the subtask's section, in our code we have the interruption's section. Here we can find, for every subtask, its own interruption clause.\n",
     "\n",
-    "Then we have some facts, like the pony's tameness assigned to 20, or the definition of a steed.\n",
+    "After the subtask's section, in our code we have the interruption's section. Here we can find, for every subtask, its own interruption clause. Then we have some known facts, like the pony's maximum possible tameness assigned to 20, or the definition of a steed. Eventually, there is a section where we store some observations like the number of carrots or saddles in our possession, or the pony's current tameness. \n",
     "\n",
-    "Eventually, there is a section where we store some observations like the number of carrots or saddles in our possession, or the pony's current tameness. \n",
+    "In the (rare) case where the agent has no better action than 'explore' and it performs 1 full map patrol, the subtask switches to riding. This was implemented to avoid endless exploring, it's a desperate last resort option."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2fa569b7",
+   "metadata": {},
+   "source": [
+    "The first subtask that is listed in the knowledge base is `action(attackEnemy)`, which can be entailed if `attack(enemy,_)` can be entailed. This counterintuitive syntax is a workaround to maintain the compatibility with the way that the `Agent.act()` method selects the other subtasks while maintaining the possibility to query which enemy to attack. This information is necessary to execute the *attackEnemy* subtask, and is better provided by the knowledge base; on the other hand the other subtasks don't have arguments tied to logic variables of any sort, since they don't need them: all the relevant information such as which is the closest element to reach is more easily provided by processing through python the information present in the knowledge base or other data structures."
```

Comment:
attackEnemy is the only clause explained and it should be the least relevant to the task ... ? i'd remove or shorten this part (saying briefly we do this for surviving purposes)